### PR TITLE
feat: integrate teamsheet logic to statsperform parsers

### PR DESCRIPTION
**Pull request by hbiermann95**
Integrate the logic of Teamsheet objects to both StatsPerform parsers. Add a seperate function to create teamsheet objects from the open statsperform files as well as seperate function to create minimal teamsheets from tracking file and a function to create more detailed teamsheets from event file. Teamsheets are used to create links between teams and team IDs and are returned in the main parsing function.

### CODE REVIEW

File: floodlight/io/statsperform.py [hunk 1]
```diff
@@ -155,55 +128,100 @@ def _read_open_event_csv_single_line(
...
+        # insert data to teamsheet
+        teamsheets[team]["player"] = [
+            pID[0] for pID in pIDs
+        ]
+        teamsheets[team]["jID"] = [
+            jID for jID in jIDs
+        ]
+        teamsheets[team]["pID"] = [
+            pID[0] for pID in pIDs
+        ]
```
Comment (draabe) [inline floodlight/io/statsperform.py:186]: is this purposely the same information as for the column "player" (two above)?

File: floodlight/io/statsperform.py [hunk 2]
```diff
@@ -821,54 +977,87 @@ def read_event_data_xml(
...
-        them to a specific xID in the respective XY objects. Should be supplied if that
-        order matters. If None is given (default), the links are automatically extracted
-        from the csv file at the cost of a second pass through the entire file.
+    filepath_tracking: str or pathlib.Path
+        Full path to the TXT file containing the tracking data.
+    teamsheet_home: Teamsheet, optional
+        Teamsheet-object for the home team used to create link dictionaries of the form
+        `links[team][jID] = xID`. The links are used to map players to a specific xID in
+        the respective XY objects. Should be supplied for custom ordering. If given as
+        None (default), teamsheet is extracted from the tracking data TXT file  and its
```
Comment (draabe) [inline floodlight/io/statsperform.py:1016]:
```suggestion
        None (default), teamsheet is extracted from the tracking data TXT file and its
```

File: floodlight/io/statsperform.py [hunk 3]
```diff
@@ -936,78 +1126,141 @@ def read_tracking_data_txt(
...
-        identified in StatsPerform files via jID, and this dictionary is used to map
-        them to a specific xID in the respective XY objects. Should be supplied if that
-        order matters. If None is given (default), the links are automatically extracted
-        from the csv file at the cost of a second pass through the entire file.
+        URL to the TXT file containing the tracking data.
+    teamsheet_home: Teamsheet, optional
+        Teamsheet-object for the home team used to create link dictionaries of the form
+        `links[team][jID] = xID`. The links are used to map players to a specific xID in
+        the respective XY objects. Should be supplied for custom ordering. If given as
+        None (default), teamsheet is extracted from the tracking data TXT file  and its
```
Comment (draabe) [inline floodlight/io/statsperform.py:1225]:
```suggestion
        None (default), teamsheet is extracted from the tracking data TXT file and its
```

**Review by draabe — CHANGES_REQUESTED** [review]

File: floodlight/io/statsperform.py (hunk 1 above)
Comment (hbiermann95) [reply floodlight/io/statsperform.py:186]: Let me know what you prefer!

Comment (draabe) [reply floodlight/io/statsperform.py:186]: I've done the latter for the Tracab teamsheet parser (that also doesn't have any player names etc. in them), but with a capital P in "Player" and indexing starting at 1, i.e.:

``player = [f"Player {i+1}" for i in range(len(jIDs))]``


I've also added the following note to the docstring:
```
    Notes
    -----
    Tracab data does not contain any player information except jersey numbers by
    default. Thus, the teamsheet objects generated by this method will name players
    'Player i' with i starting at 1. To identify players, use the jersey numbers of
    provide custom teamsheets generated by a different parser if Tracab data is used in
    combination with other data providers.
```

Comment (hbiermann95) [reply floodlight/io/statsperform.py]: Ok, I have included this structure for both parsers. But I believe there is are two words missing in the Notes docstring, "To identify players, use the jersey numbers of [players or] provide custom teamsheets..."

**Pull request merged**